{
  if (caught instanceof RuntimeException) {
    throw (RuntimeException)caught;
  }
 else {
    throw new RuntimeException(caught);
  }
}
